### DISCUSSION TITLE
GoogleOverlay not working with Google Maps Vector Maps with Google Maps Version 3.54

### DISCUSSION THREAD

**Discussion by Derek-Herr77** [visgl/deck.gl#8229] [Q&A] [answered]
When using any version of the Google Maps JS API 3.54 and above, all deck.gl layers are rendered underneath the google maps layer, making them invisible. This can be seen in Google's own documentation: [https://developers.google.com/maps/documentation/javascript/examples/deckgl-arclayer](https://developers.google.com/maps/documentation/javascript/examples/deckgl-arclayer)

This seems to only affect Vector based Google Map IDs. I have found a quick fix by setting interleaved to false like so in the GoogleMapsOverlay layer:

`const overlay = new GoogleMapsOverlay({
    interleaved: false,
    layers: [layer]
  }`

This remedies the visibility of the deck.gl layers, but they are now rendering overtop google maps labels. Any ideas on how to fix this so my google maps labels render above my deck.gl layers like in previous versions?

Note: All packages are up to date and I'm importing from @deck.gl/google-maps/typed. Tested on up-to-date Edge and Chrome browsers.

**Comment by felixpalmer** [comment]
@Derek-Herr77 I've modified the [jsfiddle](https://jsfiddle.net/gh/get/library/pure/googlemaps/js-samples/tree/master/dist/samples/deckgl-arclayer/jsfiddle) linked from the docs to use versions 3.53.13 & 3.53.14 and they work fine, only the `beta` channel is broken. Could you try to use one of those versions?

**Reply by Derek-Herr77** [reply]
Sorry, I've edited my post to make the question clearer as I incorrectly worded it. I can confirm that any version of 3.53 works perfectly, however any version 3.54 and above does not work. From my current understanding of the google maps js api, versions are deprecated after 3-6 months, so I'd like to be able to use the weekly channel, namely version 3.54.

Also update, I've found that this [jsfiddle](https://jsfiddle.net/gh/get/library/pure/googlemaps/js-samples/tree/master/dist/samples/deckgl-tripslayer/jsfiddle) works even when I render the same nonfunctional ArcLayer from the previous jsfiddle. I can confirm that this fiddle is also working on versions 3.54. I'll update if I find the cause.

**Reply by felixpalmer** [reply, marked as answer]
It seems to be down to the mapId. When I use the "New York" id of `fae05836df2dc8bb` in the "Prague" demo it works

**Reply by Derek-Herr77** [reply]
It's down to the Data-Driven Styling portion of the google maps style. If any of the feature layers are turned on for a map-id, it renders deck.gl underneath the google map. I'll just use something else in the meantime. Thanks!

**Reply by felixpalmer** [reply]
Could you file an [Issue](https://github.com/visgl/deck.gl/issues) detailing exactly how to trigger this? I imagine that it will affect more people in the future

**Reply by Derek-Herr77** [reply]
Yeah, no problem!
